test/test_contra_env_setup.py

```diff
@@ -37,73 +54,75 @@ def setUp(self):
-                ' dnf update -y nettle" --user %s --ssh-common-args="%s" ' %
-                (self.vm.username, ' '.join(ssh_args)))
-            process.run(cmd, env=env)
-        
-        if distro == 'CentOS':
-            cmd = (' ansible localhost -i "localhost," --become '
-                ' -m raw -a "yum install -y python libselinux-python git; '
-                ' yum update -y nettle" --user %s --ssh-common-args="%s" ' %
-                (self.vm.username, ' '.join(ssh_args)))
-            process.run(cmd, env=env)
-        
-        # Parameters used to DEBUG mode
-        DEBUG = self.params.get('DEBUG', default=None)
-        iso_src = self.params.get('minishift_iso_src_path', default=None)
-        iso_dest = self.params.get('minishift_iso_dest_path', default=None)
+        cmd = (' resize2fs /dev/sda1; '
+               ' {pkg_mgm} install -y python libselinux-python git; '
+               ' {pkg_mgm} update -y nettle; ').format(pkg_mgm=self.pkg_mgm)
 
-        # command use in the test from parameters
-        cmd_run = self.params.get('command', default=None)
+        if self.debug:
+            cmd += (' mkdir -p {dir}; '
+                    ' chown avocado:avocado {dir} ').format(dir=self.iso_dest)
 
-        if DEBUG is True:
-            # Create the directory to storage the ISO in VM
-            cmd = (' ansible localhost -i "localhost," '
-                   ' -m raw -a "mkdir -p %s" '
-                   ' --user %s --ssh-common-args="%s" ' %
-                   (iso_src, self.vm.username, ' '.join(ssh_args)))
-            process.run(cmd, env=env)
+        ansible_cmd = (' ansible localhost -i "localhost," --become '
+                       ' -m raw -a "{cmd}" --user {user} '
+                       ' --ssh-common-args="{ssh_args}" ')
+
+        # resize the partition
+        # install python, git and update nettle to libvirt network
+        # create the directory to storage the ISO in VM if debug enabled
+        ansible_cmd = ansible_cmd.format(cmd=cmd, user=self.vm.username,
+                                         ssh_args=' '.join(ssh_args))
+
+        process.run(ansible_cmd, env=env)
 
+        if self.debug is True:
             # Copy the minishift.iso of host to VM
             cmd = (' ansible localhost -i "localhost," '
                    ' -m copy -a "src=%s/minishift.iso dest=%s/" '
                    ' --user %s --ssh-common-args="%s" ' %
-                   (iso_src, iso_dest, self.vm.username, ' '.join(ssh_args)))
+                   (self.iso_src, self.iso_dest, self.vm.username, ' '.join(ssh_args)))
             process.run(cmd, env=env)
 
         # Run the contra-env-setup playbook
-        cmd = (' %s'
-               ' --ssh-common-args="%s"' %
-               (cmd_run, ' '.join(ssh_args)))
-        process.run(cmd, env=env)
+        if self.cmd_run is not None:
+            cmd = (' %s'
+                   ' --ssh-common-args="%s"' %
+                   (self.cmd_run, ' '.join(ssh_args)))
```

Comment:
Where is the `self.cmd_run`? I see `command: Null` in the YAML and `self.cmd_run` defaults to `None` in `setUp()`.

Reply:
in the yaml I can't set None because to yaml is a "string" and in the python, I can't check Null, in that case, I'm using None. works very well, please test to validate.